src/documentation/markdown.rs

```diff
@@ -382,4 +418,107 @@ And a line, or a rule."##;
+
+        assert_eq!(dbg!(&plain).as_str(), PLAIN);
+        assert_eq!(dbg!(&mapping).len(), 0);
+        for (reduced_range, markdown_range) in mapping.iter() {
+            assert_eq!(plain[reduced_range.clone()].to_owned(), String::from(""));
+        }
+    }
+
+    #[test]
+    fn markdown_reduction_mapping_auto_link_with_misspelled_words() {
+        const MARKDOWN: &str = r#" dyrck <http://foo.bar/baz> artic"#;
+        const PLAIN: &str = r#"dyrck  artic"#;
+
+        let (plain, mapping) = PlainOverlay::extract_plain_with_mapping(MARKDOWN);
+
+        assert_eq!(dbg!(&plain).as_str(), PLAIN);
+        assert_eq!(dbg!(&mapping).len(), 2);
+        for (reduced_range, markdown_range) in mapping.iter() {
+            assert_eq!(
+                plain[reduced_range.clone()].to_owned(),
+                MARKDOWN[markdown_range.clone()].to_owned()
+            );
+        }
+    }
+
+    #[test]
+    fn markdown_reduction_mapping_email_with_misspelled_words() {
+        const MARKDOWN: &str = r#" dyrck <joe@example.com> artic"#;
+        const PLAIN: &str = r#"dyrck  artic"#;
+
+        let (plain, mapping) = PlainOverlay::extract_plain_with_mapping(MARKDOWN);
+
+        assert_eq!(dbg!(&plain).as_str(), PLAIN);
+        assert_eq!(dbg!(&mapping).len(), 2);
+        for (reduced_range, markdown_range) in mapping.iter() {
+            assert_eq!(
+                plain[reduced_range.clone()].to_owned(),
+                MARKDOWN[markdown_range.clone()].to_owned()
+            );
+        }
+    }
+
+    #[test]
+    #[ignore]
+    fn markdown_reduction_mapping_footnote() {
+        // What is a markdown footnote syntax? Example:
+        //
+        // Here's a longer one.[^bignote]
+        // [^bignote]: Here's one with multiple paragraphs and code.
+        //
+        // From the pulldown-cmark anything can be followed in the footnote
+        // https://github.com/raphlinus/pulldown-cmark/blob/ad91999dbb0cf0ce64785ed8d22180064985b95a/src/parse.rs#L347
+        //
+        // from this test example, the event has the following form:
+        // Event: FootnoteReference(Borrowed("reference"))
+        // There are two Footnote events as we have [^reference] twice
+        // and the following text is considered as text and not part of the footnote by the parsing
+        //
+        // This followed text has no distinguishing of link or normal text.
+        // Maybe this need more thought about it as the parsing will not tell me if is a link.
+        //
+        // Also, should the reference name be checked or not? If yes, then the mapping needs to be fixed as it have some extras "[]"
+        // footnotes are always in two
+        const MARKDOWN: &str = r#" This is a footnote artic [^reference]. Which one? [^reference] ../../reference/index.html"#;
+
+        const PLAIN: &str = r#"This is a footnote artic reference. Which one? reference ../../reference/index.html"#;
```

Comment:
```suggestion
        const MARKDOWN: &str = r#"footnote [^linktxt]. Which one?
        
[^linktxt]: ../../reference/index.html"#;

        const PLAIN: &str = r#"footnote linktxt. Which one?

linktxt"#;
```

Reply:
There is a sample set for official compliance to the common mark spec here (in firefox you can expand all in that json and just search for samples) https://spec.commonmark.org/0.29/spec.json

Reply:
Thanks for the reference, I will take a look!
true, I did not really finished this cause I had the above questions that I wanted to clarify. I still would like to add another misspelled word after the reference.